PowerValidatedSolutions.psm1

```diff
@@ -50441,6 +50441,469 @@ Function Test-WMSubnetInput {
+    }
+    $fileName = Split-Path -Path $csrFilePath -LeafBase
+    $crtFileName = $fileName + ".crt"
+    $crtFilePath = Join-Path -Path $certDirPath -ChildPath $crtFileName
+    if (Test-Path -Path $crtFilePath) {
+        Write-Error "$crtFilePath file exists.  Please remove or rename the existing file." -ErrorAction Stop
+    }
+    if (-Not ([bool] (Get-Command -ErrorAction Ignore -Type Application openssl))) {
+        Write-Error "OpenSSL not found.  Please verify OpenSSLis installed: PRE_VALIDATION_FAILED" -ErrorAction Stop
+    } else {
+        $opensslVersion = openssl version | ForEach-Object { ($_ -split ' ')[1] }
+        if ($opensslVersion -lt "3.0.0") {
+            Write-Error "OpenSSL version is less than 3.0: PRE_VALIDATION_FAILED" -ErrorAction Stop
+        }
+    }
+
+    if($certificateAuthority -eq "openssl") {
+        # validate inputs
+        if (-Not (Test-Path -Path $caKeyPath)) {
+            Write-Error "Openssl CA private key file ($caKeyPath) not found: PRE_VALIDATION_FAILED" -ErrorAction Stop
+        }
+        if (-Not (Test-Path -Path $caCertPath)) {
+            Write-Error "Openssl CA certificate file ($caCertPath) not found: PRE_VALIDATION_FAILED" -ErrorAction Stop
+        }
+        if ($expireDays -lt 2) {
+            Write-Error "Certificate expiration day ($expireDays) is less than 1 day.  Please provide a value greater than 1 day."
+        }
+
+        for ($createRandomCounter = 0; $createRandomCounter -lt 8; $createRandomCounter++) {
+            $randomString = -join (((48..57)+(65..90)+(97..122)) * 80 |Get-Random -Count 8 |%{[char]$_})
+            $cfgFilePath = Join-Path -path $certDirPath -ChildPath "$randomString.cfg"
+            if (-Not(Test-Path -Path $cfgFilePath)) {
+                Write-Output "1" | Set-Content $cfgFilePath
+                if (Test-Path -Path $cfgFilePath) {
+                    Remove-Item $cfgFilePath
+                    break
+                }
+            } else {
+                if($createRandomCounter -gt 6) {
+                    Write-Error "Unable to create temporary files in directory $certDirPath." -ErrorAction Stop
+                    Exit
+                }
+            }
+        }
+
+        # Generate extension configuration file for openssl
+        $configurationText = openssl req -in $csrFilePath -text -noout
+        $configurationTextArray = $configurationText.split('`n')
+        $lineCount = 0
+        $sanArray = @()
+        foreach ($line in $configurationTextArray) {
+            if ($line -match "Subject Alternative Name:") {
+                $lineCount++
+                if($configurationTextArray[$lineCount] -match "DNS:"){
+                    $sanString = $configurationTextArray[$lineCount]
+                    $sanString = $sanString.replace('DNS:', '')
+                    $sanArray = $sanString  -split "\s*,\s*"
+                    
+                }
```

Comment:
```suggestion
                    $sanArray = $sanString  -split "\s*,\s*"
                }
```

Reply:
addressed in commit cfcd77a79eb20278e44b123b0092c173dd7b5014